impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java

```diff
@@ -553,13 +579,19 @@ protected void settings(C context, boolean emitSettingsWarnings, SettingsBuilder
                 .build();
 
         customizeSettingsRequest(context, settingsRequest);
+        if (context.eventSpyDispatcher != null) {
```

Comment:
Our regression tests fail because the `SettingsBuilderRequest` event no longer fires after this PR has been merged. Would that be because `eventSpyDispatcher` is now somehow null at this point?

Reply:
This method is called twice, once (very early) when loading up extensions, then spy is null. The 2nd invocation is when "maven boots" and then spy should not be null...

Reply:
Here's a reproducer https://github.com/alextu/maven4-reproducer